[Kyh196201] 5주차 레스토랑 조회 구현하기

kyh196201 opened the pull request:
트레이너님 안녕하세요!

리뷰 요청드립니다.

RegionsContainer와 reducer에 대한 일부 테스트 코드를 작성했고, 컴포넌트를 구현했습니다.

사실 이전 주차 두번째 과제에서 피드백 받아야할 부분 같은데, 지금 제가 구현하고 있는 순서가 맞는지 잘 모르겠습니다. 😂

kyh196201 committed:
기본 파일 구조 설정

kyh196201 committed:
given2 모듈 설치

kyh196201 committed:
Regions fixture 추가

- RegionContainer 테스트에 사용할 fixture를 추가했습니다.

kyh196201 committed:
RegionContainer 테스트 코드 작성

- 지역 목록이 있을 때 없을 떄 렌더링을 확인하는 테스트 코드를 작성했습니다.

kyh196201 committed:
RegionsContainer 컴포넌트 구현

- regions가 있을 경우 목록을 렌더링하는 코드를 구현했습니다.

kyh196201 committed:
RegionsContainer 테스트 코드 수정

- regions가 없을 경우에 대한 테스트 코드를 수정했습니다.

kyh196201 committed:
App 테스트 코드 추가

- 지역 목록이 렌더링되었는지 확인하는 테스트 코드를 추가했습니다.

kyh196201 committed:
App 컴포넌트 테스트 통과

- RegionsContainer를 App 컴포넌트 내부에서 렌더링했습니다.

kyh196201 committed:
Reducer 테스트 코드 작성

- 스토어의 초기 상태를 검증하는 테스트 코드를 작성했습니다.

kyh196201 committed:
Reducer 테스트 통과

- 인자로 전달받은 state가 없을 경우 initialState를 리턴하도록 구현했습니다.

kyh196201 committed:
Regions 컴포넌트에 대한 테스트 코드 추가

- regions props에 따라서 목록이 렌더링되었는지 확인하는 테스트코드를 작성했습니다.

kyh196201 committed:
Region 컴포넌트 테스트 통과

kyh196201 committed:
RegionsContainer 컴포넌트 리팩토링

- 엘리먼트를 렌더링하는 부분을 Regions 컴포넌트로 대체했습니다.

daadaadaah commented on src/containers/RegionsContainer.test.jsx:
```diff
@@ -0,0 +1,41 @@
+
+      const { container } = renderRegionsContainer();
+
+      const region = given.regions[0];
+
+      expect(container).toHaveTextContent(region.name);
```
regions 라는 복수형 답게 반복문으로 돌려서 여러개 확인해도 괜찮을 것 같아요!

kyh196201 replied:
넵 ! ㅎㅎ `forEach`를 이용해서 모든 regions가 렌더링되었는지 확인하는 코드로 수정했습니다 🙌

kyh196201 committed:
Reducer setRegions 테스트 코드 추가

- setRegions에 대한 테스트 코드, action creator를 추가했습니다.

kyh196201 committed:
Reducer setRegions 테스트 통과

kyh196201 committed:
Reducer에 정의되지 않은 액션에 대한 테스트 추가

kyh196201 committed:
Reducer 정의되지 않은 액션 테스트 수정

- it() 추가

kyh196201 committed:
Reducer 테스트 통과

kyh196201 committed:
서비스 추가

kyh196201 committed:
Redux thunk 로직 스토어에 추가

kyh196201 committed:
비동기 action creator 추가

kyh196201 committed:
App 컴포넌트에서 loadRegions 호출에 대한 테스트 추가

- 처음 앱이 실행될 때 loadRegions를 실행하는지 체크하는 테스트 코드를 추가했습니다.

kyh196201 committed:
App 컴포넌트 테스트 통과

- 최초 App이 렌더링될 때 loadRegions를 호출했습니다.

kyh196201 committed:
App 테스트 코드 리팩터링

- App 컴포넌트 내부에서 서비스 모듈을 import하지 않아서 불필요한 mocking을 제거했습니다.

daadaadaah commented on src/containers/RegionsContainer.test.jsx:
```diff
@@ -0,0 +1,41 @@
+jest.mock('react-redux');
+
+describe('<RegionsContainer />', () => {
+  given('regions', () => []);
```
`given`은 Context아래에 위치 하는게 코드를 이해하는 데 더 도움될 것 같아요!

```jsx
  context('without regions', () => {
    given('regions', () => []);

    it('nothing renders', () => {
      const { queryAllByRole } = renderRegionsContainer();

      expect(queryAllByRole('listitem')).toHaveLength(0);
    });
  });
```

kyh196201 replied:
앗 반드시 최초로 한 번 선언해줘야하는 줄 알았습니다 ㅎㅎ..

각 context 아래에 위치 하는게 더 의미를 파악하기 쉽네요! 감사합니다 🙏🙏

daadaadaah commented on src/components/Regions.jsx:
```diff
@@ -0,0 +1,15 @@
+      {
+        regions.map((region) => (
+          <li key={region.id}>
+            {region.name}
```
완성된 화면을 보면, 지역과 카테고리를 "클릭"하면, 해당 레스토랑이 나오는 기능이라서, button 요소가 필요할 것 같아요!ㅎㅎ

kyh196201 replied:
Region 컴포넌트를 구현하면서 수정하겠습니다 감사합니다 ㅎㅎ !!

daadaadaah commented on src/containers/RegionsContainer.test.jsx:
```diff
@@ -0,0 +1,41 @@
+    it('nothing renders', () => {
+      const { queryAllByRole } = renderRegionsContainer();
+
+      expect(queryAllByRole('listitem')).toHaveLength(0);
```
`listitem` 만 봤을 때, 어떤 리스트인지 정확히 알기 어려울 것 같아요! 어떻게 변경하는게 좋을까요?

kyh196201 replied:
이 부분은 잘 모르겠습니다 😂

Dom에 직접 접근하는 것은 안좋을 것 같고, [ByTestId](https://testing-library.com/docs/queries/bytestid#api)이 api를 사용하면 될까요??

daadaadaah reviewed (commented):
아침에 짬내서 작업 더 진행해주셨네용!! 잘하셨어요👍 commit 메시지도 작업별로 명확히 구분해주시구👍
비동기 액션들에 대한 테스트 추가해주시면서 작업해주시면 될 것 같아요!

kyh196201 committed:
RegionsContainer 테스트 리팩토링

- given regions 설정 부분을 각 context 아래에서 실행했습니다.
- forEach를 이용해서 regions에 있는 모든 목록이 렌더링되었는지 확인했습니다.

kyh196201 committed:
비동기 액션 테스트 추가

- loadRegions가 실행되면 setRegions 액션을 호출하는지 테스트하는 코드를 작성했습니다.

kyh196201 commented:
트레이너님 피드백 주신 내용 수정완료했습니다!

비동기 액션에 대한 테스트를 추가하라고 하셔서 `async-actions.js`에 있는 `loadRegions` 함수에 대한 테스트를 작성했는데, 트레이너님이 말씀하신걸 제가 맞게 이해한건지 모르겠네요 😂

그리고 `Regions` 컴포넌트 테스트에서 `listitem` role로 엘리먼트를 검색하는 방법 말고 다른 방법이 어떤게 있나요? 잘 모르겠습니다 ㅠㅠ

혹시 [ByTestId](https://haeguri.github.io/2020/12/21/clean-up-jest-mock/) api를 사용하면 될까요??

daadaadaah commented on src/containers/RegionsContainer.test.jsx:
```diff
@@ -0,0 +1,41 @@
+  const renderRegionsContainer = () => render((<RegionsContainer />));
+
+  useSelector.mockImplementation((selector) => selector({
+    regions: given.regions,
```
테스트용 데이터는 fixture에 있는 데이터를 활용하는게 좋아요!
다른 테스트에서도 마찬가지로 적용해주세요!ㅎㅎ

daadaadaah commented on src/containers/RegionsContainer.test.jsx:
```diff
@@ -0,0 +1,41 @@
+    it('renders regions', () => {
+      const { container } = renderRegionsContainer();
+
+      given.regions.forEach((region) => {
```
테스트용 데이터는 fixture에 있는 데이터를 활용하는게 좋아요!
다른 테스트에서도 마찬가지로 적용해주세요!ㅎㅎ

daadaadaah commented on src/store/async-actions.test.js:
```diff
@@ -0,0 +1,39 @@
+
+      const dispatcher = loadRegions();
+
+      await dispatcher(dispatch);
```
굳이 변수화 하지 않아도 될 것 같아요!

```jsx
      await  loadRegions()(dispatch);
```

daadaadaah commented on src/store/async-actions.test.js:
```diff
@@ -0,0 +1,39 @@
+      fetchRegions.mockRestore();
+    });
+
+    it('calls setRegions action and dispatch', async () => {
```
"dispatch setRegions" 이렇게 간단히 써주셔도 될 것 같아요!

daadaadaah commented on src/store/async-actions.test.js:
```diff
@@ -0,0 +1,39 @@
+  });
+
+  describe('loadRegions', () => {
+    afterEach(() => {
+      fetchRegions.mockRestore();
+    });
+
+    it('calls setRegions action and dispatch', async () => {
+      fetchRegions.mockResolvedValue(REGIONS);
```
BeforeEach로 관리하는게 더 좋을 것 같아요!
```jsx
      beforeEach(() => {
        fetchRegions.mockResolvedValue(REGIONS);
      });
```

daadaadaah reviewed (commented):
잘하고 계십니당😄  간단하게 "레스토랑 지역 보여주기" 기능으로 큰 흐름을 먼저 피드포워드 받으면서 과제 방향성 체크하시는 거 보면, 똑똑하신 것 같아요👍👍👍 더운날 지치지마시고~ 과제 완성까지 같이 힘내봅시당💪 그리고! 테스트 코드 작성하실 때, Happy path에 대해서 먼저 작성하는 거 잊지마시구요!! 다음 PR 기대할게요🔥

> 그리고 Regions 컴포넌트 테스트에서 listitem role로 엘리먼트를 검색하는 방법 말고 다른 방법이 어떤게 있나요? 잘 모르겠습니다 ㅠㅠ
혹시 [ByTestId](https://haeguri.github.io/2020/12/21/clean-up-jest-mock/) api를 사용하면 될까요??

간단하게 "지역이름이 안보인다" 라고 해도 괜찮고, 
```jsx
     REGIONS.forEach((region) => {
        expect(container).not.toHaveTextContent(region.name);
      });
```

만약 실제 서비스라면, regions이 없는 상황이 어떤 경우인지에 대해 생각해보면, api 통신이 원활하게 이루어지지 않은 경우라서, 사용자에게 "지역을 불러오지 못했습니다." 등이 보인다라고 해도 괜찮을 것 같아요!

daadaadaah commented on src/store/async-actions.test.js:
```diff
@@ -0,0 +1,39 @@
+
+      await dispatcher(dispatch);
+
+      expect(dispatch).toBeCalledWith(setRegions(REGIONS));
```
`redux-mock-store`를 사용해서 실제로 어떤 `actions`을 호출하는지 확인할 수 있어요! 한번 사용해봅시당!
참고 : https://github.com/reduxjs/redux-mock-store

kyh196201 committed:
Async-action 테스트 코드 리팩터링

- 불필요한 변수 dispatcher를 제거했습니다.
- loadRegions 테스트 설명을 수정했습니다.
- afterEach 대신 beforeEach 라이프사이클을 이용해서 fetchRegions의 리턴값을 설정했습니다.

kyh196201 committed:
RegionsContainer 테스트 수정

- 지역 목록이 없을 경우 메시지가 렌더링되었는지 확인하는
테스트 코드를 작성했습니다.

kyh196201 committed:
Regions 테스트 코드 수정

- 지역 목록이 없을 경우에 대한 테스트 코드를 수정했습니다.

kyh196201 committed:
Regions, RegionsContainer 테스트 통과

kyh196201 committed:
Region 테스트 추가

- region name이 렌더링되었는지에 대한 테스트 코드를 작성했습니다.

kyh196201 committed:
Region 컴포넌트 테스트 통과

kyh196201 committed:
Regions 컴포넌트 리팩터링

- Region 컴포넌트를 내부에 렌더링하도록 수정했습니다.

kyh196201 committed:
Region 테스트 코드 수정

- getByText로 검사할 경우 텍스트와 완전 일치할 경우만 찾기 때문에
 toHaveTextContent 로 변경했습니다.

kyh196201 committed:
Region 테스트 코드 추가

- 클릭 이벤트에 대한 테스트 코드를 추가했습니다.

kyh196201 committed:
Region 컴포넌트 테스트 통과

- onClick props를 추가했습니다.

kyh196201 commented:
트레이너님 오늘은 시간이 부족해서 피드백 주신 내용 수정하고, 테스트 몇개만 추가했습니다 😂

내일 코멘트에 남겨주신 라이브러리 적용하고, 추가 테스트 및 기능 구현하겠습니다 감사합니다 🙌

daadaadaah commented on src/components/Region.test.jsx:
```diff
@@ -1,11 +1,17 @@
-  const renderRegion = (region) => render((<Region region={region} />));
+  const handleClick = jest.fn();
+
+  const renderRegion = (region) => render((<Region region={region} onClick={handleClick} />));
```
사소한 거지만 props가 여러 개일 경우, 세로 정렬이 읽기 더 편한 것 같아요!

daadaadaah reviewed (commented):
잘했습니당! 바쁘신 일정 중에 조금씩이라도 PR 날려주시는거 너무 좋아요👍
계속 진행해주세요🔥🔥🔥

kyh196201 committed:
Regions 테스트 추가

- 지역 버튼 클릭에 대한 테스트 코드를 추가했습니다.

kyh196201 committed:
Regions 컴포넌트 테스트 통과

- onClickRegion props를 추가했습니다.

kyh196201 committed:
selectRegion 액션 추가

kyh196201 committed:
RegionContainer 테스트 추가

- 지역 버튼 클릭에 대한 테스트를 추가했습니다.

kyh196201 committed:
RegionsContainer 컴포넌트 테스트 통과

- 지역 버튼 클릭 시 selectRegion 액션을 dispatch하도록 구현했습니다.

kyh196201 committed:
Reducer 테스트 추가

- selectRegion 액션에 대한 테스트를 추가했습니다.

kyh196201 committed:
Reducer 테스트 통과

- reducer initialState에 regionId의 초기값을 설정했습니다.
- reducer에 selectRegion 액션에 대한 로직을 추가했습니다.

kyh196201 committed:
Region 테스트 추가

- 선택되었을 경우에 대한 테스트를 추가했습니다.

kyh196201 committed:
Region 테스트 통과

- 선택되었는지를 판단하기 위해 selected props를 추가했습니다.
- 선택되었을 경우 (V) 문구를 표시하도록 수정했습니다.

kyh196201 committed:
Region 테스트 추가

- 선택되어있지 않을 경우에 대한 테스트를 추가했습니다.

kyh196201 committed:
Regions 테스트 추가

- 선택된 지역이 있을 경우, 없을 경우에 대한 테스트를 추가했습니다.

kyh196201 committed:
Regions 테스트 통과

- selectedRegionId를 props로 받고,
Region 컴포넌트에 selected props를 계산해서 전달했습니다.

kyh196201 commented:
트레이너님 지역 관련된 테스트 코드랑 기능 구현 거의다 완료해가네요! 🙌

그리고 질문이 있습니다.

---

1. 선택된 지역인지 판단하는 로직이 어디에 위치하면 좋을지 궁금합니다.

Region 컴포넌트는 `selected` props를 전달받고,
이 `selected`는 Regions 컴포넌트에서 계산해서 내려주고 있는데, Region 컴포넌트 내부에서 선택되었는지를 판단하는게 더 좋을까요??

2. Regions 컴포넌트가 `selectedRegionId`라는 props를 받고 있는데, state에서는 `regionId`라고 선언해서 사용하고 있습니다. 
Regions 컴포넌트에는 선택된 지역 ID라는 의미를 강조하고 싶어서 props 명을 저렇게 설정했는데, 두 변수명을 맞추는게 좋을까요??

3. 선택된 지역이 있을 경우 '(V)' 문구가 렌더링되었는지 확인하기 위해서 Regions와 Region에서 아래처럼 테스트 코드를 작성했습니다.
혹시 더 좋은 방법이 있는지 트레이너님의 생각이 궁금합니다!

```javascript
expect(container).toHaveTextContent('(V)');
```

daadaadaah commented on src/components/Regions.jsx:
```diff
@@ -0,0 +1,29 @@
+          <li key={region.id}>
+            <Region
+              region={region}
+              selected={isSelected(region)}
```
props 네이밍은 `isSelected` 이고, function 네이밍은 `selectRegion`이런건 어떠신가요?

kyh196201 replied:
감사합니다! 항상 이벤트 핸들러를 props로 전달할 때 `on`을 붙여야된다는 고정관념?이 있었는데 `selectRegion`이 훨씬 의미도 드러나고 좋은 것 같습니다 감사합니다 🙌

daadaadaah replied:
제가 의도한 건 `isSelected`을 `selectRegion`변경하는 걸 말하는 거였는데, 설명이 부족했네용😅

daadaadaah replied:
`onSelectRegion`을 의도한건대ㅎㅎ제가 설명이 부족했네용😂 다음에는 좀더 명확하게 코멘트드리도록 하겠습니당!

kyh196201 replied:
아닙니다 트레이너님 ㅎㅎ 항상 좋은 피드백 너무 감사해요😊

daadaadaah commented on src/store/reducer.js:
```diff
@@ -0,0 +1,16 @@
+
+export const initialState = {
+  regions: [],
+  regionId: null,
```
그냥 `regionId`라고 하면, 어떨 때 필요한 `regionId`이지? 라는 생각이 들어요!
좀더 의도를 드러내봅시당!ㅎㅎ

daadaadaah commented on src/containers/RegionsContainer.jsx:
```diff
@@ -0,0 +1,24 @@
+  return (
+    <>
+      <Regions
+        regions={regions}
+        onClickRegion={handleClickRegion}
```
`selectedRegionId` props로 redux의 `regionId`를 활용해보시는 건 어떠신가요?

kyh196201 replied:
앗 이 부분도 아직 구현하지 못한 코드였는데, 추가했습니다 감사합니다 ㅎㅎ

daadaadaah reviewed (changes requested):
지역 관련 코드가 거의 완성되어가고 있네요!👍 남은 주말에 다른 기능도 완성시켜봅시당!🚀🚀🚀

그리고, 컴포넌트나 함수 만드실 때, `default function parameter` 많이 활용하시는 것 같은데, 자신만의 기준을 가지고 활용해보시는게 좋을 거 같아요! redux에서의 state 초기값이랑 UI 컴포넌트에서의 `default function parameter`의 초기값을 깜박하고 다르게 넣어주면, 버그가 발생할 수도 있을 것 같아서요!


> 1. 선택된 지역인지 판단하는 로직이 어디에 위치하면 좋을지 궁금합니다.
Region 컴포넌트는 `selected` props를 전달받고, 이 `selected`는 Regions 컴포넌트에서 계산해서 내려주고 있는데, Region 컴포넌트 내부에서 선택되었는지를 판단하는게 더 좋을까요??

- 말씀하시는 선택된 지역인지 판단하는 로직이 아래 코드 말씀하시는거 맞나용?ㅎㅎ
```jsx
const isSelected = (region) => region.id === selectedRegionId;
```
- 어떤 로직이 어디에 있으면 좋을지를 판단할 때에는 각 컴포넌트의 관심사가 무엇인지에 대해 생각해보고 결정할 것 같아요! 지금 Region 컴포넌트는 그냥 자기가 선택 받았는지 여부만 알면 되므로, 그 상태만 props로 내려주면 될 것 같아요! 또한, 이와 관련해서 redux에서 `regionsId`로 선택된 region을 관리하고 있으니, 이 상태를 활용해보세요!


> 2. Regions 컴포넌트가 `selectedRegionId`라는 props를 받고 있는데, state에서는 `regionId`라고 선언해서 사용하고 있습니다. 
Regions 컴포넌트에는 선택된 지역 ID라는 의미를 강조하고 싶어서 props 명을 저렇게 설정했는데, 두 변수명을 맞추는게 좋을까요??

- selectedRegionId로 sync를 맞춰주는 것이 좋을 것 같아요! state에서 `regionId`만 있으면, 언제 필요한 상태인지라는 의문이 생겨요!

> 3. 선택된 지역이 있을 경우 '(V)' 문구가 렌더링되었는지 확인하기 위해서 Regions와 Region에서 아래처럼 테스트 코드를 작성했습니다.
혹시 더 좋은 방법이 있는지 트레이너님의 생각이 궁금합니다!

```javascript
expect(container).toHaveTextContent('(V)');
```

그냥 `(V)` 표시보다 `서울(V)` 이런식으로 구체적으로 적는게 좋을 것 같아요!

kyh196201 committed:
State 네이밍 수정

regionId를 선택되었다는 의도를 드러내기 위해서,
selectedRegionId로 변경했습니다.

kyh196201 committed:
Region 컴포넌트 수정

- (V)문구와 지역 이름 사이에 공백을 제거했습니다.

kyh196201 committed:
Region, Regions 테스트 코드 수정

- 선택된 지역인지 확인하는 코드를 수정했습니다.

kyh196201 committed:
RegionsContainer 테스트 수정

- 선택된 지역이 있을 경우, 없을 경우에 대한 테스트를 추가했습니다.

kyh196201 committed:
RegionsContainer 테스트 통과

- selectedRegionsId를 Regions 컴포넌트에 props로 전달했습니다.

kyh196201 committed:
Region 컴포넌트 리팩토링

- selected props의 네이밍을 isSelected로 수정했습니다.

kyh196201 committed:
Regions 컴포넌트 리팩터링

kyh196201 committed:
Regions 컴포넌트 리팩터링

- onClickRegion props의 네이밍을 selectRegion으로 수정했습니다.

kyh196201 committed:
카테고리 테스트를 위한 fixture 데이터 추가

kyh196201 committed:
Category 테스트 추가

kyh196201 committed:
Category 테스트 통과

kyh196201 committed:
Categories 테스트 추가

kyh196201 committed:
Categories 테스트 통과

kyh196201 committed:
카테고리 관련 action 추가

kyh196201 committed:
CategoriesContainer 테스트 추가

kyh196201 committed:
CategoriesContainer 테스트 통과

kyh196201 committed:
Reducer initialState 수정

- 카테고리 목록, 선택된 카테고리 ID를 추가했습니다.

kyh196201 committed:
Reducer 테스트 추가

- setCategories, selectCategory애 대한 테스트를 추가했습니다.

kyh196201 committed:
Reducer 테스트 통과

- 추가한 액션에 대한 로직을 구현했습니다.

kyh196201 committed:
loadCategories 테스트 추가

kyh196201 committed:
loadCategories 구현

kyh196201 committed:
App 테스트 수정

- 최초 카테고리 목록을 조회했는지에 대한 테스트를 추가했습니다.

kyh196201 committed:
App 테스트 통과

- 최초 렌더링 시 카테고리 목록을 조회하도록 수정했습니다.

kyh196201 committed:
App 테스트 수정

- 카테고리 목록이 있을 경우 렌더링했는지에 대한 테스트를 추가했습니다.

kyh196201 committed:
App 컴포넌트 수정

- CategoriesContainer 컴포넌트를 추가했습니다.

kyh196201 committed:
레스토랑 테스트를 위한 fixture 추가

kyh196201 committed:
RestaurantsContainer 테스트 추가

kyh196201 committed:
RestaurantsContainer 테스트 통과

kyh196201 committed:
Restaurants 테스트 추가

kyh196201 committed:
Restaurants 테스트 통과

kyh196201 committed:
RestaurantsContainer 리팩터링

- 내부 UI에 관련된 로직을 제거하고, Restaurants 컴포넌트를 렌더링했습니다.

kyh196201 committed:
Restaurant 테스트 추가

kyh196201 committed:
Restaurant 테스트 통과

kyh196201 committed:
Restaurants 리팩터링

- 각 레스토랑을 렌더링하는 엘리먼트를 Restaurant 컴포넌트로 대체했습니다.

kyh196201 committed:
App 테스트 수정

- 레스토랑 목록이 있을 경우, 없을 경우에 대한 테스트를 추가했습니다.

kyh196201 committed:
App 테스트 통과

- RestaurantsContainer 컴포넌트를 추가했습니다.

kyh196201 committed:
App 테스트 리팩터링

- regions, categories가 없을 경우에 대한 테스트를 추가했습니다.

kyh196201 committed:
Reducer 테스트 수정

- setRestaurants 액션에 대한 테스트를 추가했습니다.

kyh196201 committed:
setRestaurants 액션 추가

kyh196201 committed:
Reducer 테스트 통과

- 초기 상태에 restaurants를 추가했습니다.
- setRestaurants 액션에 대한 로직을 구현했습니다.

kyh196201 committed:
loadRestaurants 테스트 추가

kyh196201 committed:
loadRestaurants 테스트 통과

kyh196201 committed:
RestaurantsContainer 테스트 추가

- 선택된 지역, 카테고리가 있을 경우 레스토랑을 조회하는 것에 대한
테스트를 추가했습니다.

kyh196201 committed:
RestaurantsContainer 테스트 통과

- 선택된 지역, 카테고리가 변경되면 레스토랑을 조회하도록 구현했습니다.

kyh196201 committed:
Categories 커버리지 테스트 통과

kyh196201 committed:
RestaurantsContainer 테스트 커버리지 통과

kyh196201 committed:
actions 리팩터링

- 불필요한 공백을 제거했습니다.

kyh196201 committed:
api 테스트 코드 추가

kyh196201 commented:
트레이너님 이번 한 주도 너무 감사했습니다! 😊

피드백 주신 내용 적용하고, 나머지 추가 기능 구현했는데, 커버리지 테스트가 아직 통과를 못했네요 이 부분은 마저 리팩토링 진행하겠습니다!!

알려주신 `redux-mock-store`는 결국 적용을 못해봤습니다 ㅠㅠ

kyh196201 committed:
Regions 컴포넌트 테스트 커버리지 통과

kyh196201 committed:
App 컴포넌트 given 초기 값 설정

kyh196201 committed:
레스토랑 e2e 테스트 수정

kyh196201 committed:
지역 선택 시 id가 아니라 region 객체를 넘기도록 수정

kyh196201 committed:
지역 선택 로직 수정

kyh196201 committed:
Regions 테스트 수정

- 선택된 지역 ID가 아닌 선택된 지역을 통해 테스트하도록 수정했습니다.

kyh196201 committed:
Regions 테스트 통과

- selectedRegion을 props로 받도록 수정했습니다.

kyh196201 committed:
RegionsContainer 테스트 수정

- selectedRegionId 대신 selectedRegion을 사용하도록 수정했습니다.

kyh196201 committed:
RegionsContainer 테스트 통과

kyh196201 committed:
loadRestaurants 테스트 수정

- selectedRegion을 인자로 전달받아서 api 요청을 하는지에 대한
테스트를 추가했습니다.

kyh196201 committed:
loadRestaurants 테스트 통과

kyh196201 committed:
RestaurantsContainer 테스트 수정

kyh196201 committed:
RestaurantsContainer 테스트 통과

kyh196201 committed:
Categories 예외처리 코드 추가

daadaadaah commented on src/App.test.jsx:
```diff
@@ -0,0 +1,121 @@
+import App from './App';
+
+jest.mock('react-redux');
+jest.mock('./store/async-actions');
```
이렇게 mocking 해주실 때, ` __mocks__`폴더 하나 만들어서 mocking하는 함수 넣어주시고, 테스트 코드에서 mockImplement 해주셔야됩니당!ㅎㅎ

daadaadaah reviewed (commented):
이번주도 수고하셨어요! 이번주 돌아보면서 주간회고 한번 작성해보세요~ 😃
